src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -266,5 +268,20 @@ public void Build(IListedCapabilityStatement statement)
                 statement.TryAddRestInteraction(SystemRestfulInteraction.Transaction);
             }
         }
+
+        public void Dispose()
+        {
+            _transactionScope?.Dispose();
+        }
+
+        public void BeginTransactionScope()
```

Comment:
I think we should follow more closely with the .net patterns of, so in the datastore it only needs 1 method for setup:
```csharp
ITransaction BeginTransaction()
```
ITransaction could have a `SqlTransaction` implementation.

ITransaction has the other methods:
```csharp
public interface ITransaction : IDisposable {
void Complete();
void Dispose();
}
```

In calling code, it would then be easy to wrap in a using:

```csharp
using(var transaction = _transactionHandler.BeginTransaction()) {
// do things
transaction.Commit();
}
```
Any errors would call Dispose without Commit and cause a rollback

Thoughts?

Reply:
Done as per suuggestion.